rpc/send.go

```diff
@@ -208,7 +209,16 @@ func Send(opts Options, method string, addrs []net.Addr, getArgs func(addr net.A
 // client is ready. On success, the reply is sent on the channel;
 // otherwise an error is sent.
 func sendOne(client *Client, timeout time.Duration, method string, args, reply interface{}, c chan interface{}) {
-	<-client.Ready
+	if timeout == 0 {
+		// Wait forever.
+		timeout = math.MaxInt64
+	}
+	select {
+	case <-client.Ready:
```

Comment:
now that I think about it, you also need to listen for the <-client.Closed (which is what gets triggered when the connect retry loop gives up)